### ISSUE TITLE
Blazor Tooling: .razor-based Components are not recognized by Intellisense until corresponding razor file is opened (reopen of #19188)

### ISSUE THREAD

**Issue by fitdev** [dotnet/aspnetcore#22169]
Please see #19188 for background. Since that issue was closed and cannot be reopened by me I am posting an update.

Now I am using latest VS Preview: VS 2019 16.7 Preview 1, with .Net 5 Preview 4.

I was able to determine the following...

### CSProj.PartialLoadMode

VS Feature Flag called `CSProj.PartialLoadMode` was set to `true`. When I set it to `false`, _most_ of the issues went away and _most_ of the components' tags were recognized as such by Razor syntax highlighting.

### _Imports.razor

The rest (as far as I could test) of the component tags not being recognized as such by Razor syntax highlighting were resolved by me manually adding appropriate `@using Namespace` directives into _all_ `_Imports.razor` along the directory tree.

Now, this, as far as I understand, should _not_ be necessary, since the `@using Namespace` directives are additive: child (nested in the subfolders) `_Imports.razor` files simply add to the list of already imported namespaces by the "parent" `_Imports.razor` files. As far as compilation stage and runtime stage are concerned, this is indeed how it works (as there was never an error about missing using). However, as far as tooling (syntax highlighting and intellisense) is concerned, I was forced to duplicate the `@using Namespace` in all the _child_ `_Imports.razor` files.

Hopefully these 2 pieces of information are enough for you to address the issue now.

**Comment by fitdev** [comment]
UPDATE: It also appears now that the problem returns after I do some basic Razor-Component-related edits in any one of the files.

So, initially (upon VS restart and Solution load), once I have made the 2 changes I described above, everything is recognized and highlighted correctly.

However after I have edited just a bit - and just 1 razor file by changing its "shape" - like added another parameter for example, then VS is quite busy (CPU working hard for 20-30 seconds) and many of the Razor component tags are once again not recognized in the other razor files (document tabs).

It is only when I close those tabs, reopen them again do the corresponding Razor components start being recognized by VS. And I have to do this procedure from the _"bottom up"_ - beginning to reopen razor components starting with the one that has no dependencies on (does not contain) other razor components, working my way up to the razor components with more component dependencies.

A few things to reiterate:
- This applies only for design-time experience. Everything compiles fine and runs fine. So likely some-kind of design0-time code gen issue.
- Only applies to components defined in `.*razor` files. Those components defined _entirely_ in `*.cs` files are always highlighted properly.
